### PR TITLE
Automate updating of README to reflect current packages

### PR DESCRIPTION
**Pull request by amichaelparker**
### Summary
Per @bjankord's comment on [issue 1146](https://github.com/cerner/terra-core/issues/1146), this uses the npm package 'markdown-magic' to automatically update the README.md with the current packages state.

### Additional Details
markdown-magic is setup to crawl through the packages folder and update the README.md with the packages listed in the tree, as well as providing the usual links and status tags.

Run 'npm run docs' to test.

Since deprecated packages have been deleted, the currently listed ones have been moved manually to a section below the auto-generated list.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -108,6 +106,16 @@ Components in beta stage may include breaking changes, new features, and bug fix
...
+<!-- AUTO-GENERATED-CONTENT:END *-->
+
+### Deprecated
+
+| Terra Package      | Version | Status | Dependencies |
+|--------------------|---------|--------|--------------|
+| terra-application | [![NPM version](https://badgen.net/npm/v/terra-application)](https://www.npmjs.org/package/terra-application) | ![Deprecated](https://badgen.net/badge/status/Deprecated/grey) | ![terra-application](https://badgen.net/badge/dependencies/n%2Fa/grey) |
+| [terra-form](https://github.com/cerner/terra-core/tree/master/packages/terra-form) | [![NPM version](https://badgen.net/npm/v/terra-form)](https://www.npmjs.org/package/terra-form) | ![Deprecated](https://badgen.net/badge/status/Deprecated/grey) | ![terra-form](https://badgen.net/badge/dependencies/n%2Fa/grey) |
```
Comment (bjankord) [inline README.md]: We can unlink the terra-form link at the beginning of this table row.

Comment (amichaelparker) [reply README.md]: Link removed.

File: markdown.config.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+          }
+          return entry;
+        })
+        .join('\n');
+    },
+  },
+};
+
+const markdownPath = path.join(__dirname, 'README.md');
+markdownMagic(markdownPath, config);
```
[the remark this answers is not in the document]
Comment (amichaelparker) [reply markdown.config.js]: Works for me in Windows 10 via command prompt / powershell.

**Comment by yuderekyu** [comment]
have any thoughts on linking this script with the [generator-terra-module](https://github.com/cerner/generator-terra-module#generated-package)? it would make sense to generate this after scaffolding

**Comment by bjankord** [comment]
@yuderekyu I'm not sure how we could link the generator with this script, but I'm open to ideas. I could see us adding some doc around when creating a new package, run this script after you run the generator.

**Comment by bjankord** [comment]
Yeah, plan is to move this README generation code to framework and clinical as well. Let's track running this after running the generator in a new issue. I'll log one for us to follow up on.

**Pull request merged**